### PR TITLE
[harfbuzz] Use meson port

### PR DESCRIPTION
**Pull request by ebraminio**
As our migration to meson, pretty much shaped after ots's, let's see how it goes

### CODE REVIEW

**Comment by ebraminio** [comment]
Is dataflow bot fail, `clang-11: error: no such file or directory: '/usr/lib/DataFlow*.o'`, related to this change?

**Comment by ebraminio** [comment]
Oh some other fails I have no idea :/

**Comment by Dor1s** [comment]
dataflow one shouldn't be related at all, let me see what's up with the recent images -- maybe something's changed in llvm or base builder

**Comment by Dor1s** [comment]
Actually, llvm and base images are fine. Could it be so that either meson or ninja is not seeing those DataFlow object files? In the same log you can see the following line:

```
Compiling DataFlow to /usr/lib/DataFlow*.o...  done.
```

which confirms that OSS-Fuzz side of things isn't broken

**Comment by ebraminio** [comment]
Just done a null edit on the port and even more things fail #4082 :) understandably complicated situation you got there, what about #4081 so you'll be in full control of compiler arguments you pass, am not seriously proposing it but that somehow makes reproductions also easier for me.

**Comment by Dor1s** [comment]
Hm, MSan failure somehow mentions ndpi project, while harfbuzz seems to be alright: https://github.com/google/oss-fuzz/pull/4076/checks?check_run_id=843035910

**Comment by Dor1s** [comment]
@ebraminio do you know why the dataflow objects path is getting passed with single quotes?

```
clang++  -o test/fuzzing/hb-shape-fuzzer 'test/fuzzing/5325d79@@hb-shape-fuzzer@exe/hb-shape-fuzzer.cc.o' -Wl,--as-needed -Wl,--no-undefined -O1 -fno-omit-frame-pointer -gline-tables-only -DFUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION -fsanitize=dataflow -fsanitize-coverage=trace-pc-guard,pc-table,bb,trace-cmp -stdlib=libc++ -fno-sanitize=vptr -DHB_NO_VISIBILITY -Wl,--start-group src/libharfbuzz.a src/libharfbuzz-subset.a '/usr/lib/DataFlow*.o' -pthread -lm -Wl,--end-group '-Wl,-rpath,$ORIGIN/../../src' -Wl,-rpath-link,/work/build/src
clang-11: error: no such file or directory: '/usr/lib/DataFlow*.o'
```

That's why dataflow build is failing, the `*` inside single quotes isn't expanding into actual file paths.

**Comment by Dor1s** [comment]
> Hm, MSan failure somehow mentions ndpi project, while harfbuzz seems to be alright: https://github.com/google/oss-fuzz/pull/4076/checks?check_run_id=843035910

@inferno-chromium have you seen such behavior before?

File: projects/harfbuzz/build.sh [hunk 1]
```diff
@@ -17,31 +17,24 @@
...
-    $LIB_FUZZING_ENGINE ./src/.libs/libharfbuzz.a
+# cleanup
+rm -rf $build
+mkdir -p $build
 
-$CXX $CXXFLAGS -std=c++11 -Isrc \
-    ./test/fuzzing/hb-subset-fuzzer.cc -o $OUT/hb-subset-fuzzer \
-    $LIB_FUZZING_ENGINE ./src/.libs/libharfbuzz-subset.a ./src/.libs/libharfbuzz.a
+# Build the library.
+meson --default-library=static -Dfuzzer_ldflags=$LIB_FUZZING_ENGINE \
```
Comment (Dor1s) [inline projects/harfbuzz/build.sh:31]: if you don't find a solution for getting rid of the single quotes, you can do something like

```
-Dfuzzer_ldflags="$(echo $LIB_FUZZING_ENGINE)"
```

Comment (ebraminio) [reply projects/harfbuzz/build.sh:31]: Oh, let me try it.

Comment (ebraminio) [reply projects/harfbuzz/build.sh]: now turned to https://github.com/google/oss-fuzz/pull/4076/checks?check_run_id=847216709#step:7:525 which hmm, makes sense somehow.

Comment (ebraminio) [reply projects/harfbuzz/build.sh]: will add a split https://github.com/harfbuzz/harfbuzz/blob/master/test/fuzzing/meson.build#L17 here, that should help I guess

Comment (ebraminio) [reply projects/harfbuzz/build.sh]: https://github.com/harfbuzz/harfbuzz/commit/e04050e3b856d275779a2d010fc59ff230a8c4d8 let's see

Comment (ebraminio) [reply projects/harfbuzz/build.sh]: worked very well, thank you very much :) it makes sense also, we should apply that for ots also.

**Comment by ebraminio** [comment]
All green now, I can't believe it :) weird on `cla`, I've already have contributions here and on Chromium and kubernetes helm https://github.com/helm/helm/pull/1019 and so.

File: projects/harfbuzz/build.sh (hunk 1 above)
Comment (ebraminio) [reply projects/harfbuzz/build.sh]: > if you don't find a solution for getting rid of the single quotes

It makes sense I believe (or at least nothing we can in harfbuzz can improve AFAIK), guess meson likes to help on things may become wildcard by quoting them.

Comment (ebraminio) [reply projects/harfbuzz/build.sh]: https://mesonbuild.com/FAQ.html#why-cant-i-specify-target-files-with-a-wildcard

**Comment by Dor1s** [comment]
CLA looks good: the commits are authored by your @gnu.org email which has CLA signed for. Not sure why the CLA bot got stuck, I'm merging this anyways.

**Comment by ebraminio** [comment]
Thank you so much! :)

**Pull request merged**